### PR TITLE
[Functions] Can't create functions with m-TLS 

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

#7255 re-worked Function MetaDataManager to make all metadata writes only by the leader.
This unintentionally broke Pulsar Functions when m-TLS is used for authentication. Because it doesn't
taken TLS port into consideration and always uses a non-TLS port to communicate with the leader broker.

The PR fixes the broken implementation and ensure Pulsar Functions use the right service url and
authentication plugin to communicate with leader.

*Tests*

Add an integration test to reproduce the issue and ensure functions worker with m-TLS

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by codelipenghui — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]

**Review by freeznet — APPROVED** [review]
LGTM

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java [hunk 1]
```diff
@@ -1316,19 +1323,24 @@ private void startPackagesManagementService() throws IOException {
...
         String hostname = ServiceConfigurationUtils.getDefaultOrConfiguredAddress(
             brokerConfig.getAdvertisedAddress());
         workerConfig.setWorkerHostname(hostname);
-        workerConfig.setWorkerPort(brokerConfig.getWebServicePort().get());
         workerConfig.setWorkerId(
             "c-" + brokerConfig.getClusterName()
                 + "-fw-" + hostname
-                + "-" + workerConfig.getWorkerPort());
+                + "-" + (workerConfig.getWorkerPort() != null
+                    ? workerConfig.getWorkerPort() : workerConfig.getWorkerPortTls()));
```
Comment (klwilson227) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1340]: workerConfig.getTlsEnabled() ? workerConfig.getWorkerPort(): workerConfig.getWokerPortTls()

I believe would be consistent with the usage of TLS. Infering from the fact that the config has setup the port or not may lead to differences between the usage and the naming...

**Review by klwilson227 — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java (hunk 1 above)
Comment (sijie) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java:1340]: Sure. can make that change.

Comment (sijie) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java]: Fixed the logic to make it consistent.

**Review by klwilson227 — CHANGES_REQUESTED** [review]
I believe in there is a change missing in that from within the Worker.java start() method, there is a order of operations issue in which the classes initialize() method is called at 69 which then creates the PulsarAdmin class passing workerConfig, but then the workerService.start() method is called which completes the setup of the workerConfig... Thus the PulsarAdmin is initialized with incomplete configuration.

**Review by jerrypeng — APPROVED** [review]

**Review by jiazhai — APPROVED** [review]

**Comment by jiazhai** [comment]
looks like the tls auth failed in the test:
```
14:14:38.240 [docker-java-stream-161014990:org.apache.pulsar.tests.integration.utils.DockerUtils$2@257] ERROR org.apache.pulsar.tests.integration.utils.DockerUtils - DOCKER.exec(PulsarFunctionsThreadTest-vvxth-pulsar-functions-worker-thread-lxgbp-1:/pulsar/bin/pulsar-admin functions status --tenant public --namespace default --name test-tumbling-window-fn-zbjaarbg): completed with non zero return code: 1
stdout: 
stderr: org.apache.pulsar.client.admin.PulsarAdminException: java.util.concurrent.CompletionException: org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector$RetryException: Could not complete the operation. Number of retries has been exhausted. Failed reason: error:10000410:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE

Reason: HTTP 500 Internal Server Error
```

**Comment by zymap** [comment]
I use another PR #9553 for fixing this on branch 2.7. So I will remove the `release/2.7.1` label.

**Pull request merged**